src/main/java/klerer/rijks/RijksSearchFrame.java

```diff
@@ -0,0 +1,157 @@
+import klerer.rijks.json.ArtObjects;
+
+import javax.imageio.ImageIO;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.io.IOException;
+import java.net.URL;
+
+public class RijksSearchFrame extends JFrame {
+    private RijksService rijksService;
+    private JTextField searchBar;
+    private JButton prevButton;
+    private JButton nextButton;
+    private JPanel imagePanel;
+    private int currentPage = 1;
+    private ApiKey apiKey = new ApiKey();
+    private String keyString = apiKey.get();
+    private Disposable disposable;
+
+    public RijksSearchFrame() {
+        setTitle("Rijks Museum");
+        setSize(1050, 850);
+        setDefaultCloseOperation(EXIT_ON_CLOSE);
+        setLayout(new BorderLayout());
+
+        searchBar = new JTextField(60);
+        prevButton = new JButton("Previous Page");
+        nextButton = new JButton("Next Page");
+        JPanel top = new JPanel();
+        top.add(prevButton);
+        top.add(searchBar);
+        top.add(nextButton);
+        add(top, BorderLayout.NORTH);
+
+        imagePanel = new JPanel(new GridLayout(2, 5, 5, 5));
+        add(new JScrollPane(imagePanel), BorderLayout.CENTER);
+
+        rijksService = new RijksServiceFactory().getService();
+
+        prevButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                if (currentPage > 1) {
+                    currentPage--;
+                    performSearch();
+                }
+            }
+        });
+
+        nextButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                currentPage++;
+                performSearch();
+            }
+        });
+
+        searchBar.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                currentPage = 1;
+                performSearch();
+            }
+        });
+
+        performSearch();
+    }
+
+    private void performSearch() {
+
+        if (searchBar.getText().isEmpty()) {
+            disposable = rijksService.pageNum(keyString, currentPage)
+                    // tells Rx to request the data on a background Thread
+                    .subscribeOn(Schedulers.io())
+                    // tells Rx to handle the response on Swing's main Thread
+                    .observeOn(SwingSchedulers.edt())
+                    //.observeOn(AndroidSchedulers.mainThread()) // Instead use this on Android only
+                    .subscribe(
+                            (response) -> handleResponse(response),
+                            Throwable::printStackTrace);
+        } else {
+            disposable = rijksService.queryAndPageNum(keyString, searchBar.getText(), currentPage)
```

Comment:
Use the other method on rijksService to search for the involvedMaker

Reply:
I should add an else if or I should change line 93?

Reply:
Change the line. The search might work

Reply:
I tried changing it but its not working. It works to search an artist but not anything else and after the first search its very slow

Reply:
> I tried changing it but its not working. It works to search an artist but not anything else and after the first search its very slow

I thought you said artist search wasn't working

Reply:
it wasn't but now artist search is working and its not working to search anything else